src/Event/EuLoginEventSubscriber.php

```diff
@@ -67,12 +67,33 @@ public static function create(ContainerInterface $container) {
    */
   public static function getSubscribedEvents(): array {
     $events = [];
-    $events[CasHelper::EVENT_PRE_REGISTER] = 'processUserProperties';
+    $events[CasHelper::EVENT_PRE_REGISTER] = [
+      ['checkUserMailExists', 1000],
+      ['processUserProperties'],
+    ];
     $events[CasHelper::EVENT_POST_VALIDATE] = 'processCasAttributes';
     $events[CasHelper::EVENT_PRE_VALIDATE] = 'alterValidationPath';
     return $events;
   }
 
+  /**
+   * Checks user e-mail exists previously.
+   *
+   * @param \Drupal\cas\Event\CasPreRegisterEvent $event
+   *   The triggered event.
+   */
+  public function checkUserMailExists(CasPreRegisterEvent $event): void {
+    $cas_settings = $this->configFactory->get('cas.settings');
+    if ($cas_settings->get('user_accounts.auto_register')) {
+      $email = $event->getCasPropertyBag()->getAttribute('email');
+
+      if (user_load_by_mail($email) !== FALSE) {
```

Comment:
Don't need the `!== FALSE`, if a value is tested as a boolean, PHP will cast it to a boolean. In this case, if `user_load_by_mail()` returns a UserInterface object it will cast it to `TRUE`.